### PR TITLE
[XPIPELINE-97] Polish HDRP Global Settings

### PR DESCRIPTION
**Pull request by jenniferd-unity**
### Purpose of this PR
This PR aims to finish and polish the implementation of HDRP Global Settings for 2021.2.
Here is a list of changes:

Code-only changes:
- Refactored code to use m_globalSettings instead of HDRenderPipelineGlobalSettings.instance for partial class of HDRenderPipeline [suggested by Julien Ignace]
- Removed internal API HDRenderPipeline.defaultAsset
- Renamed SerializedFrameSettings defaultFrameSettings;  to defaultCameraFrameSettings
- Cleaned a bit HDRenderPipeline localized strings (removed unnecessary ones)
- Renamed VolumeProfileLookDev to LookDevVolumeProfile - as suggested by Remi Slysz

Fixed broken fixes:
- Wizard - profile creation: default [PR 3873](https://github.com/Unity-Technologies/Graphics/pull/3873) - previously fixed by Julien Ignace
- DXR - missing resources depending on Quality Level: [1320304](https://fogbugz.unity3d.com/f/cases/1320304/) - previously fixed by Anis Benyoub
-  [1313426](https://fogbugz.unity3d.com/f/cases/1313426)

Fixes:
- HDRP Global Settings Asset deletion when HDRP is active or not (the asset is recreated only if HDRP is active + an WARNING message displays in the console)
image
(Example showing error message when switching from URP to HDRP and no HDRP Global Settings asset was assigned in the Graphics Settings)
- Warnings and Info messages in the HDRP Settings menu are clearer
image
- Fixed HDRP Asset creation when HDRP Global Settings was not present (for ex. my project was on URP only)
- Frame Settings refactor (see below)
- incl. fix for [https://fogbugz.unity3d.com/f/cases/1323947/](https://fogbugz.unity3d.com/f/cases/1323947/) 

image

Frame Settings refactor (see [here](https://unity.slack.com/archives/C6Y79CZM0/p1616689251241700)):
- Renamed in the HDRP Settings window Frame Settings to Frame Settings (Default Values)
- made sure the value displayed in the Custom Frame Settings showed the value of the HDRP Global Settings asset
- Changed: you can always edit a setting no matter the active HDRP Asset
- Changed: when you edit Custom Frame Settings, you can always override values no matter the parent or HDRP Asset. it is disabled only when you cannot change them because it has no impact.
- Changed: Virtual Texturing setting is no longer disabled in the UI but sanitized instead, better tooltip to show 
- Improved a few tooltips - still more to do but will be another PR
- Checked the Debug Render Pipeline menu to ensure values shown are still correct

Example of how Frame Settings behave now (Camera / HDRP Global Settings):
frame-settings-behavior

Example of changes:
image

---
### Testing status
Tested on Windows Editor for local test project and Template HD.
Tried also a build of Template HD.

Known issues: https://fogbugz.unity3d.com/f/cases/1323947/
Important to test that broken fixes are actually fixed + Frame Settings behave as expected

---
### Comments to reviewers
This is the last PR for HDRP Global Settings planned before cut-off.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/Lighting/Reflection/ReflectionMenuItem.cs
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/Lighting/Reflection/ReflectionMenuItem.cs:21]: So we switch from default (gfx one) to current (potentially quality). Is this ok ?

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs:228]: Why it should be Medium above all other possibility? I mean if defaultFrameSettings is null, it means we are displaying the version in the global settings. So why it should be linked to Medium quality mode? This seams strange. At least it need comments to explain why we chose this default here.

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Skin.cs
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Skin.cs:11]: Should this be localized?

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/OverridableFrameSettingsArea.cs
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/OverridableFrameSettingsArea.cs:166]: uneeded parenthesis

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/SerializedHDRenderPipelineGlobalSettings.cs
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/SerializedHDRenderPipelineGlobalSettings.cs:102]: ha sorry but now we have another name inconsistency here:
**LookDev**VolumeProfile
VolumeProfile**Default**

Should this be defaultVolumeProfile?

File: com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/DefaultSettingsVolumeProfile.asset
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/DefaultSettingsVolumeProfile.asset:47]: Be extra cautious with asset modification. I am not sure you wanted to change this.

File: com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/DefaultLookDevProfile.asset
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/DefaultLookDevProfile.asset:73]: here too: value changed. Please check

File: com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/DefaultLookDevProfile.asset
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/DefaultLookDevProfile.asset:76]: here too: value changed. Please check

File: com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/DefaultSettingsVolumeProfile.asset
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/DefaultSettingsVolumeProfile.asset:97]: here too: value changed. Please check

File: com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/DefaultSettingsVolumeProfile.asset
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/DefaultSettingsVolumeProfile.asset:261]: here too: value changed. Please check

File: com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/DefaultSettingsVolumeProfile.asset
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/DefaultSettingsVolumeProfile.asset:267]: here too: value changed. Please check

File: com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/DefaultSettingsVolumeProfile.asset
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/DefaultSettingsVolumeProfile.asset:39]: also for all added entry (here and bellow), check if it is still the default from code

File: com.unity.render-pipelines.high-definition/Editor/Wizard/HDWizard.Configuration.cs
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/Wizard/HDWizard.Configuration.cs:442]: it is not an hdrpAsset anymore. Please rename the variable

Comment (jenniferd-unity) [reply com.unity.render-pipelines.high-definition/Editor/Wizard/HDWizard.Configuration.cs:442]: fixed

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/SerializedHDRenderPipelineGlobalSettings.cs
Comment (RSlysz) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/SerializedHDRenderPipelineGlobalSettings.cs]: Is there any release with this version landed?
yes -> yes you need to
no -> just change it quickly before any release land

**Comment by RSlysz** [comment]
Better :)
Check value in asset change and add a comment on why default on MaterialQualityMode.Medium and it will be good :)

File: com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/DefaultSettingsVolumeProfile.asset
Comment (jenniferd-unity) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipelineResources/DefaultSettingsVolumeProfile.asset:97]: will remove this file from the PR - i am not supposed to change it

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/DefaultSettingsPanel.cs
Comment (jenniferd-unity) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/DefaultSettingsPanel.cs:87]: Note for reviewers: Currently error and warning messages in C# are left unlocalized.
See https://confluence.unity3d.com/display/LOC/Editor+Localization+Coding+Guidelines

File: com.unity.render-pipelines.high-definition/Editor/Lighting/Reflection/ReflectionMenuItem.cs
Comment (jenniferd-unity) [reply com.unity.render-pipelines.high-definition/Editor/Lighting/Reflection/ReflectionMenuItem.cs:21]: i could actually rewrite that to grab the HDRP Global Settings one!

Comment (jenniferd-unity) [reply com.unity.render-pipelines.high-definition/Editor/Lighting/Reflection/ReflectionMenuItem.cs:21]: fixed in upcoming commit

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs
Comment (jenniferd-unity) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs:228]: we needed to chose a value, i didnt know which one to use so i chose medium

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/SerializedHDRenderPipelineGlobalSettings.cs
Comment (jenniferd-unity) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/SerializedHDRenderPipelineGlobalSettings.cs]: those changes landed in srp2core so i did a migration pass in updated file

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/DefaultSettingsPanel.cs
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/DefaultSettingsPanel.cs:157]: I think code convetion does not allow '_' inside names

Comment (jenniferd-unity) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/DefaultSettingsPanel.cs:157]: The rest of the file is like that and other windows UI code, so it should be good no?

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs
Comment (RSlysz) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs:228]: I think this need to be checked, perhaps ask TA and UX for this point.

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/DefaultSettingsPanel.cs
Comment (RSlysz) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/DefaultSettingsPanel.cs:157]: Well, according to code convention, we should not use _ except for prefixing https://ono.unity3d.com/unity-extra/unity-meta/raw/@/ReferenceSource/CSharp/Assets/CSharpReference.cs

So I assum it was already wrong at the time :/

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs
Comment (RSlysz) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.Drawers.cs:228]: But it can be fixed in a later PR perhaps

**Review by RSlysz — APPROVED** [review]
Seams good. Check with UX/TA for default on medium quality for material. This need to be clarified (can be in later PR)

**Review by JordanL8 — APPROVED** [review]

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs
Comment (eh-unity) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs:458]: Maybe just use break in the if instead of second condition of !requiresRayTracingResources
I think it's a bit easier to read that way.

**Review by eh-unity — APPROVED** [review]
LGTM.

**Review by iM0ve — COMMENTED** [review]
**Testing: In progress**

**Whats tested:**
* Regression testing
  * Creating a new HDRP project and running the DXR wizard
  * Using Frame settings on multiple cameras
  * Frame settings can override Default settings. ex. If you turn on Opaque on camera but turn it off in Default - opaque objects will be rendered.
  * Switching quality settings Low->Med->High on Template. Works as intended.
  * Debug Scene camera and Other cameras correctly display active Frame Settings
  * If something is Off in the HDRP asset, but On in Frame settings. The feature will not be present in the scene, as expected.
  * Real-world usage: Upgrading Amalienborg, changing Frame Settings
* Every fix as requested in PR
  *  Wizard - profile creation: default PR 3873 - previously fixed by Julien Ignace
  * Building DXR HDRP template on High preset
  * Null ref when URP and HDRP are installed [1313426](https://fogbugz.unity3d.com/f/cases/1313426)
  * HDRP Global Settings Asset is no longer created when URP or Default is the current renderer
  * HDRP Default settings not saving, case [1323947](https://fogbugz.unity3d.com/f/cases/1323947/)
* Frame Settings refactor
  * Frame Settings is renamed
  * You can always edit a setting no matter the active HDRP Asset
  * you edit Custom Frame Settings, you can always override values no matter the parent or HDRP Asset. it is disabled only when you cannot change them because it has no impact.

**Issues blocking the PR:**
1) **Fixed**  Frame settings can not be saved

2) **Fixed** Error on reflection probe change

https://user-images.githubusercontent.com/36502659/114169727-9ca2cd00-993a-11eb-85d7-6698eadd35eb.mp4



**Issues I found that I will file fogbugz for, not blocking the PR:**
1)  The path no longer exists described in the warning

image

2) Global settings ignore the name specified in HDRP wizard

https://user-images.githubusercontent.com/36502659/113277420-5bcb0880-92e9-11eb-8225-d858d4ae9f0b.mp4

**Comment by iM0ve** [comment]
The PR mentions:
incl. fix for https://fogbugz.unity3d.com/f/cases/1323947/

But I still cant seem to be able to save the Frame settings. Is this the same issue? They get lost on domain reload

https://user-images.githubusercontent.com/36502659/113425659-f2c3bd80-93da-11eb-9524-f75fcc983e7f.mp4


https://user-images.githubusercontent.com/36502659/113425664-f5261780-93da-11eb-9894-9946151aa045.mp4

**Comment by jenniferd-unity** [comment]
Submitted fixes for the 3 issues reported @iM0ve and synced to latest master yesterday :) 
Changes reviewed by @RSlysz too.
will relaunch a yamato but we should be good for another QA pass

**Review by iM0ve — APPROVED** [review]
Updated the review above. Should be good to merge now. 
Filed 2 fogbugz for cosmetic issues:
https://fogbugz.unity3d.com/f/cases/1327978/
https://fogbugz.unity3d.com/f/cases/1327981/

**Pull request merged**
